rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,181 @@
+inspecting the commit messages -- using `git log`, for instance.
+
+Because commit messages are an indirect but important aspect of code quality,
+and also important for code maintenance, it is essential for a long term open
+source project to ensure that they meet high standards.
+
+The importance of commit messages conveying enough context and information about
+the code being changed will grow as the project grows and bad (poorly written)
+commit messages can affect negatively the code quality of future changes that
+would otherwise benefit from past good commit messages if they existed.
+
+Beyond code itself, poorly written commit messages can also affect the community
+in other ways. For example, by not providing to new contributors a consistent
+and complete history or context for the code changes, it can work as a barrier
+for new contributions because much more time will be necessary trying to
+understand what motivated a past critical but unclear change.
+
+Hence this Commit Message Guideline can help contributors to write good commit
+messages and so improve the current situation regarding the TVM commit logs.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Commit Message Guideline
+
+Apache TVM uses the Github (GH) platform for patch submission and code review
+via Pull Requests (PRs). The final commit (title and body) that is merged into
+the Apache TVM main tree is composed of the PR's title and body and must be kept
+updated and reflecting the new changes in the code as per the reviews and
+discussions.
+
+Although these guidelines apply essentially to the PRs’ title and body messages,
+because GH auto-generates the PR’s title and body from the commits on a given
+branch, it’s recommended to follow these guidelines right from the beginning,
+when preparing commits in general to be submitted to the Apache TVM project.
+This will ease the creation of a new PR, avoiding rework, and also will help the
+review.
+
+The rules below will help to achieve uniformity that has several benefits, both
+for review and for the code base maintenance as a whole, helping you to write
+commit messages with a good quality suitable for the Apache TVM project,
+allowing fast log searches, bisecting, and so on.
+
+_PR/commit title_:
+
+* Guarantee a title exists (enforced);
+* Don’t use Github usernames in the title, like @username (enforced);
+* Check if a tag should be present as a hint about what component(s) of the code
+  the commits “touch”. For example [BugFix], [CI], [microTVM], and [TVMC]. Tags
+  go between square brackets and appear first in the title. Tags help reviewers
+  to identify the PRs they can/want to review and also help the release folks
+  when generating the release notes (enforced);
+* Use an imperative mood. Avoid titles like “Added operator X” and “Updated
+  image Y in the CI”, instead use the forms “Add feature X” and “Update image Y
+  in the CI” instead;
```

Comment:
Would simple present tense not also work? "Adds feature X," "updates image Y," etc. Is the use of the imperative a firmly established norm?

Reply:
In my experience, I don't recall about the use of present tense being used in the commit message title. The imperative mood is the best choice in my opinion and is the common practice on lots of open source projects that I can think of.

Reply:
@slyubomirsky the intention when proposing this guidance, is to **enforce** (_meaning it blocks a PR to be merged_) the really minimal that we don't have today.

That's why we don't want to be 100% prescriptive with enforced rules. Let's get the basics right (meaning we have commit messages that are more meaningful than e.g. `fix bug`, `cleanup` or `lint code`) and then we can evolve this gradually.

Based on this, let us know if you're ok to proceed this way (Marking this as resolved when you agree), or you have any other question.

Reply:
@slyubomirsky friendly ping.

Reply:
@leandron Since @slyubomirsky but a thumbs up on my comment I'm mark it as resolved. Feel free to reopen it if that's not the case. Thanks.